programs/sig-aggregator/src/aggregator.rs

```diff
@@ -50,53 +59,105 @@ impl Aggregator for AggregatorService {
+            *cached_height = aggregate_response.clone();
+        }
+
+        Ok(Response::new(aggregate_response))
+    }
+}
+
+impl AggregatorService {
+    /// Query all attestors concurrently and collect successful responses
+    async fn query_all_attestors(
+        &self,
+        min_height: u64,
+    ) -> Result<Vec<AttestationsFromHeightResponse>> {
         let mut futs = FuturesUnordered::new();
-        let timeout_ms = Duration::from_millis(self.config.attestor.attestor_query_timeout_ms);
+        let timeout_duration = Duration::from_millis(self.config.attestor_query_timeout_ms);
 
-        // Spin up one future per client, each with its own timeout
+        // Create futures for all attestor queries
         for client in self.attestor_clients.iter() {
             let mut client = client.lock().await;
             let req = Request::new(AttestationsFromHeightRequest { height: min_height });
+
             futs.push(async move {
-                match timeout(timeout_ms, client.get_attestations_from_height(req)).await {
+                match timeout(timeout_duration, client.get_attestations_from_height(req)).await {
                     Ok(Ok(resp)) => Ok(resp.into_inner()),
-                    Ok(Err(status)) => Err(status),
-                    Err(_) => Err(Status::deadline_exceeded("attestor RPC timed out")),
+                    Ok(Err(status)) => Err(status.into_aggregator_error()),
+                    Err(_) => Err(AggregatorError::timeout(
+                        self.config.attestor_query_timeout_ms,
+                    )),
                 }
             });
         }
 
-        let mut attestator_data = AttestatorData::new();
-        while let Some(res) = futs.next().await {
-            match res {
-                Ok(att_resp) => attestator_data.insert(att_resp),
-                Err(e) => tracing::error!("An attestor query failed: {}", e),
-            }
-        }
+        let mut successful_responses = Vec::new();
+        let mut error_count = 0;
 
-        if let Some(agg_resp) = attestator_data.get_latest(self.config.attestor.quorum_threshold) {
-            let mut cached_height = self.cached_height.write().await;
-            if cached_height.height < agg_resp.height {
-                *cached_height = agg_resp;
+        while let Some(result) = futs.next().await {
+            match result {
+                Ok(response) => successful_responses.push(response),
+                Err(e) => {
+                    error_count += 1;
+                    tracing::error!("Attestor query failed: {e}");
+                }
             }
         }
 
-        let cached_height = self.cached_height.read().await;
-        if cached_height.height < min_height {
-            return Err(Status::not_found(format!(
-                "No valid attestation found for height >= {min_height}",
+        if successful_responses.is_empty() {
+            return Err(AggregatorError::internal(format!(
```

Comment:
What is the thinking behind this error type instead of having a specific error for this?